src/Build/ElementLocation/XmlDocumentWithLocation.cs

```diff
@@ -340,7 +347,7 @@ private void DetermineWhetherToLoadReadOnly(string fullPath)
                 {
                     _loadAsReadOnly = true;
                 }
-                else if (s_readOnlyFlags == ReadOnlyLoadFlags.LoadReadOnlyIfAppropriate)
+                else if (s_readOnlyFlags == ReadOnlyLoadFlags.LoadReadOnlyIfAppropriate && fullPath is object)
```

Comment:
nit: That's an unusual way to check for null. First I was like 🤔 and then 🤦 and 🙃

Reply:
I blame @jaredpar: https://twitter.com/jaredpar/status/1171474157667643400

Reply:
This is **the** way to check for non-null 😉

Reply:
This comment string was an emotional roller coaster for me. First an email from a MSBuild PR where @rainersigwald said "i blame @jaredpar" which is downright terrifying.  To actually looking at the PR and seeing proper `null` check code :smile:

Reply:
I guess I haven't been keeping up with the latest development in C#. The way to check for non-null for some, operator abuse for others 😛 

Is it part of a larger effort to make C# read like English? Trying to help VB developers switch the language? Genuinely curious, thank you!

Reply:
Found the answer on Twitter. Cool stuff!